jvm-agent/src/main/org/jetbrains/lincheck/jvm/agent/TraceAgentParameters.kt

```diff
@@ -83,6 +85,9 @@ object TraceAgentParameters {
 
     @JvmStatic
     lateinit var classUnderTraceDebugging: String
+    
+    @JvmStatic
+    lateinit var classUnderTraceDebuggingMethodOwner: String
```

Comment:
Now there are two fileds with the same purpose `classUnderTraceDebugging` and  `classUnderTraceDebuggingMethodOwner`, and they are used inconsistently throughout codebase. 

Can we please leave only `classUnderTraceDebugging`.

Reply:
It seems they are not the same because replacing one with another would lead to calling all the ancestors of the base test class.

Reply:
> It seems they are not the same because replacing one with another would lead to calling all the ancestors of the base test class.

Let's preserve the old behavior for the cases when `classUnderTestDebugging` is the declaring class of the method. 

For the cases when `classUnderTestDebugging` is not the declaring method, let's set `classUnderTestDebugging` to be the declaring class. 

All usages of `classUnderTestDebugging` should remain the same --- it is the class we are instrumenting.  

Currently there is a lot of confusion, because sometimes in transformers we lookup for `classUnderTraceDebuggingMethodOwner` as a class to instrument, and sometimes for `classUnderTestDebugging`.